lib/coxir/struct/channel.ex

```diff
@@ -160,6 +162,146 @@ defmodule Coxir.Struct.Channel do
     API.request(:post, "channels/#{channel}/messages/bulk-delete", %{messages: messages})
   end
 
+  @doc """
+  Modifies the name of a given channel.
+
+  Returns a channel object upon success
+  or a map containing error information.
+  """
+  @spec set_name(channel, String.t) :: map
+
+  def set_name(%{id: id}, name),
+    do: set_name(id, name)
+
+  def set_name(channel, name) do
+    edit(channel, %{name: name})
+  end
+
+  @doc """
+  Modifies the topic of a given channel.
+
+  Returns a channel object upon success
+  or a map containing error information.
+  """
+  @spec set_topic(channel, String.t) :: map
+
+  def set_topic(%{id: id}, topic),
+    do: set_topic(id, topic)
+
+  def set_topic(channel, topic) do
+    edit(channel, %{topic: topic})
+  end
+
+  @doc """
+  Enables the NSFW status of a given channel.
+
+  Returns a channel object upon success
+  or a map containing error information.
+  """
+  @spec enable_nsfw(channel) :: map
+
+  def enable_nsfw(%{id: id}),
+    do: enable_nsfw(id)
+
+  def enable_nsfw(channel) do
+    edit(channel, %{nsfw: true})
+  end
+
+  @doc """
+  Disables the NSFW status of a given channel.
+
+  Returns a channel object upon success
+  or a map containing error information.
+  """
+  @spec disable_nsfw(channel) :: map
+
+  def disable_nsfw(%{id: id}),
+    do: disable_nsfw(id)
+
+  def disable_nsfw(channel) do
+    edit(channel, %{nsfw: false})
+  end
+
+  @doc """
+  Modifies the position of a given channel.
+
+  Returns a channel object upon success
+  or a map containing error information.
+  """
+  @spec set_position(channel, Integer.t) :: map
+
+  def set_position(%{id: id}, position),
+    do: set_position(id, position)
+
+  def set_position(channel, position) do
+    edit(channel, %{position: position})
+  end
+
+  @doc """
+  Modifies the bitrate of a given voice channel.
+
+  Returns a channel object upon success
+  or a map containing error information.
+  """
+  @spec set_bitrate(channel, Integer.t) :: map
+
+  def set_bitrate(%{id: id}, bitrate),
+    do: set_bitrate(id, bitrate)
+
+  def set_bitrate(channel, bitrate) do
+    if bitrate != 128000 and bitrate < 8000 and bitrate > 96000 do
```

Comment:
I would remove this and let Discord return an error instead.
This way we avoid having to *hardcode* these values.